umh-core/pkg/config/mock.go

```diff
@@ -16,17 +16,21 @@ package config
 
 import (
 	"context"
+	"fmt"
 	"sync"
 	"time"
+
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/models"
 )
 
 // MockConfigManager is a mock implementation of ConfigManager for testing
 type MockConfigManager struct {
-	GetConfigCalled bool
-	Config          FullConfig
-	ConfigError     error
-	ConfigDelay     time.Duration
-	mutex           sync.Mutex
+	GetConfigCalled   bool
+	Config            FullConfig
+	ConfigError       error
+	ConfigDelay       time.Duration
+	mutexReadOrWrite  sync.Mutex
```

Comment:
please explain the mutexes, use RWMutex, etc. as discussed in our meeting jsut now